### PR TITLE
Modify preinst macos

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

update preinst macos script

### Describe how to test/QA your changes

- Install the Agent on MacOS and check the installation is successful
- 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: omnibus/package-scripts/agent-dmg/postinst [hunk 1]
```diff
@@ -57,11 +57,11 @@ chmod -v 755 "$RUN_DIR"
 echo "# Copying conf"
 mkdir -vp $CONF_DIR/checks.d
 
-if [ -e "/tmp/datadog.conf" ] || [ -e "/tmp/datadog.yaml" ]; then
+if [ -e "$INSTALL_DIR/tmp/datadog.conf" ] || [ -e "$INSTALL_DIR/tmp/datadog.yaml" ]; then
     mv -vf /tmp/datadog.conf /tmp/datadog.yaml $CONF_DIR
```
Comment (Pythyu) [inline omnibus/package-scripts/agent-dmg/postinst:61]: ❓ Question :
Shouldn't we move this one as well from `$INSTALL_DIR/tmp/datadog.conf` ?

**Review by Pythyu — COMMENTED** [review]
❓ Question :
if `$INSTALL_DIR` is `/opt/datadog-agent/`, I don't see how some of these file are created like  `$INSTALL_DIR/tmp/datadog.conf` AFAIK there's no `tmp` folder in `/opt/datadog-agent/` ?

File: omnibus/package-scripts/agent-dmg/postinst (hunk 1 above)
Comment (KevinFairise2) [reply omnibus/package-scripts/agent-dmg/postinst:61]: Yes you are right, I forgot this one. Thanks!

**Review by zgdatadoghq — APPROVED** [review]
lgtm

**Comment by KevinFairise2** [comment]
> ❓ Question : if `$INSTALL_DIR` is `/opt/datadog-agent/`, I don't see how some of these file are created like `$INSTALL_DIR/tmp/datadog.conf` AFAIK there's no `tmp` folder in `/opt/datadog-agent/` ?

The temporary files mentioned in the `postinst` script are created in the `preinst` script. Mainly to save existing conf and restore it when installing the agent on the host that already has the agent installed

**Review by chouetz — APPROVED** [review]
ok

**Pull request merged**